service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java

```diff
@@ -86,55 +91,96 @@ void allPipelinesHaveDefinedInputs() {
   }
 
   @Test
-  void allPipelineInputsAreProperlyTyped() {
+  void allPipelineInputDefinitionsAreProperlyTyped() {
     // make sure all pipeline inputs have defined types matching the enum
     for (PipelineInputDefinition p : pipelineInputDefinitionsRepository.findAll()) {
       assertDoesNotThrow(() -> PipelineInputTypesEnum.valueOf(p.getType()));
     }
   }
 
+  @Test
+  void allDefaultValuesForPipelineInputsAreCorrectType() {
+    // make sure all pipeline input definition default values pass type validation and are cast-able
+    for (PipelineInputDefinition p : pipelineInputDefinitionsRepository.findAll()) {
+      if (p.getDefaultValue() != null) {
+        PipelineInputTypesEnum inputType = PipelineInputTypesEnum.valueOf(p.getType());
+        assertNull(inputType.validate(p.getName(), p.getDefaultValue()));
+        assertNotNull(inputType.cast(p.getName(), p.getDefaultValue(), new TypeReference<>() {}));
+      }
+    }
+  }
+
   @Test
   void imputationPipelineHasCorrectInputs() {
     // make sure the imputation pipeline has the correct inputs
     Pipeline pipeline = pipelinesRepository.findByName(PipelinesEnum.IMPUTATION_BEAGLE.getValue());
 
     List<PipelineInputDefinition> pipelineInputDefinitions = pipeline.getPipelineInputDefinitions();
 
-    // currently we have one input for the imputation pipeline
-    assertEquals(1, pipelineInputDefinitions.size());
+    // currently we have five inputs for the imputation pipeline
+    assertEquals(5, pipelineInputDefinitions.size());
+
+    // there should be one user-provided input and 4 service-provided inputs
+    assertEquals(
+        1,
+        pipelineInputDefinitions.stream().filter(PipelineInputDefinition::getUserProvided).count());
+    assertEquals(
+        4,
+        pipelineInputDefinitions.stream()
+            .filter(Predicate.not(PipelineInputDefinition::getUserProvided))
+            .count());
 
-    PipelineInputDefinition input1 = pipelineInputDefinitions.get(0);
-    assertEquals("multi_sample_vcf", input1.getName());
-    assertEquals(PipelineInputTypesEnum.VCF.toString(), input1.getType());
-    assertTrue(input1.getIsRequired());
-    assertNotNull(input1.getId());
+    assertEquals(
+        "multi_sample_vcf",
+        pipelineInputDefinitions.stream()
+            .filter(PipelineInputDefinition::getUserProvided)
+            .findFirst()
+            .get()
+            .getName());
```

Comment:
i think you can follow the same pattern as you did below so its easier to add to in the future.  Test for a set of one is still good.